test/reduce.cpp

```diff
@@ -523,6 +523,221 @@ TEST(MinMax, NaN)
     }
 }
 
+TEST(MinMax, MinLocNaN)
+{
+    float test_data[] = { 1, NAN, 5, 0.1, NAN, -0.5, NAN, 0 };
+    int rows = 4;
+    int cols = 2;
+    af::array a(rows, cols, test_data);
+
+    float gold_min_val[] = { 0.1, -0.5 };
+    int gold_min_idx[] = { 3, 1 };
+    
+    af::array min_val;
+    af::array min_idx;
+    af::min(min_val, min_idx, a);
+
+    std::vector<float> h_min_val(cols);
+    min_val.host(&h_min_val[0]);
+
+    std::vector<int> h_min_idx(cols);
+    min_idx.host(&h_min_idx[0]);
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_FLOAT_EQ(h_min_val[i], gold_min_val[i]);
+    }
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_FLOAT_EQ(h_min_idx[i], gold_min_idx[i]);
+    }
+}
+
+TEST(MinMax, MaxLocNaN)
+{
+    float test_data[] = { 1, NAN, 5, 0.1, NAN, -0.5, NAN, 0 };
+    int rows = 4;
+    int cols = 2;
+    af::array a(rows, cols, test_data);
+
+    float gold_max_val[] = { 5.0, 0.0 };
+    int gold_max_idx[] = { 2, 3 };
+    
+    af::array max_val;
+    af::array max_idx;
+    af::max(max_val, max_idx, a);
+
+    std::vector<float> h_max_val(cols);
+    max_val.host(&h_max_val[0]);
+
+    std::vector<int> h_max_idx(cols);
+    max_idx.host(&h_max_idx[0]);
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_FLOAT_EQ(h_max_val[i], gold_max_val[i]);
+    }
+
+    for (int i = 0; i < cols; i++) {
+        ASSERT_FLOAT_EQ(h_max_idx[i], gold_max_idx[i]);
+    }
+}
+
+TEST(MinMax, MinCplxNaN)
+{
+    float real_wnan_data[] = {
+        849.088836, NAN, 889.189274, NAN, 506.472337,
+        NAN, NAN, 106.103029, 128.447592, 672.343090
+    };
+
+    float imag_wnan_data[] = {
+        NAN, NAN, 728.556795, 741.470103, 465.926785,
+        175.408103, NAN, 464.726170, NAN, 934.297652
+    };
+
+    int rows = 5;
+    int cols = 2;
+    af::array real_wnan(rows, cols, real_wnan_data);
+    af::array imag_wnan(rows, cols, imag_wnan_data);
+    af::array a = af::complex(real_wnan, imag_wnan);
+
+    float gold_min_real[] = { 506.472337, 106.103029 };
+    float gold_min_imag[] = { 465.926785, 464.726170 };
+    
+    af::array min_val = af::min(a);
+
+    std::vector< std::complex<float> > h_min_val(cols);
```

Comment:
you can define the symbols you use using the statements. Like this:
```
using std::vector;
using std::complex;
etc.
```
Then you don't have to specify the namespace.

Reply:
Thanks, this should eliminate some namespace bloat